app/controllers/application_controller/ci_processing.rb

```diff
@@ -1874,6 +1874,12 @@ def vm_button_operation(method, display_name, partial_after_single_selection = n
           @single_delete = true unless flash_errors?
         end
 
+        # Reset session[:snap_selected] if that snapshot is being deleted
+        if method == 'remove_snapshot' && session[:snap_selected].present? &&
+           Snapshot.find(session[:snap_selected]).vm_or_template_id.to_s == params[:id]
```

Comment:
@bmclaughlin can't we just check for existence of Snapshot record here like this: Snapshot.find_by_id(session[:snap_selected]).nil?